runtime/development/src/evm.rs

```diff
@@ -0,0 +1,113 @@
+// Copyright 2021 Centrifuge Foundation (centrifuge.io).
+//
+// This file is part of the Centrifuge chain project.
+// Centrifuge is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version (see http://www.gnu.org/licenses).
+// Centrifuge is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+use cfg_primitives::{MAXIMUM_BLOCK_WEIGHT, NORMAL_DISPATCH_RATIO};
+use codec::{Decode, Encode};
+use frame_support::{parameter_types, traits::FindAuthor, weights::Weight, ConsensusEngineId};
+use pallet_evm::{AddressMapping, EnsureAddressTruncated};
+use sp_core::{crypto::ByteArray, H160, U256};
+use sp_runtime::{traits::AccountIdConversion, Permill};
+use sp_std::marker::PhantomData;
+
+use crate::{AccountId, Aura};
+
+pub mod precompile;
+
+pub struct FindAuthorTruncated<F>(PhantomData<F>);
+impl<F: FindAuthor<u32>> FindAuthor<H160> for FindAuthorTruncated<F> {
+	fn find_author<'a, I>(digests: I) -> Option<H160>
+	where
+		I: 'a + IntoIterator<Item = (ConsensusEngineId, &'a [u8])>,
+	{
+		if let Some(author_index) = F::find_author(digests) {
+			let authority_id = Aura::authorities()[author_index as usize].clone();
+			return Some(H160::from_slice(&authority_id.to_raw_vec()[4..24]));
+		}
+		None
+	}
+}
+
+#[derive(Encode, Decode, Default)]
+struct EthereumAccount(H160);
```

Comment:
Since this module is called `evm` I'd suggest we rename this to just `Account` or `Address`:

```suggestion
struct Account(H160);
```